### DISCUSSION TITLE
oci-hook and /etc/hosts

### DISCUSSION THREAD

**Discussion by jontte75** [containers/podman#20147] [Q&A] [answered]
I have a need to check /etc/hosts file in oci-hook and in some cases even make changes to the hosts file. This actually worked fine while using slirp4netns, but after changing to pasta (it looks like that at least) it seems that /etc/hosts is updated much later or then some issue with the namespace.

So the hook is in stage "createRuntime", but tried also poststart.

The path where I'm looking for is got from config container root  and looks about like this:
`/home/contuser/.local/share/containers/storage/overlay/<a hash>/merged/etc/hosts`

I even tried to create a file to /etc/ in stage createRuntime, and that worked fine. The hook is joined to containers mnt namespace.

Should this even work? Is there a way to read and edit /etc/hosts using a oci-hook, when pasta is used?

**Comment by Luap99** [comment, marked as answer]
There is no difference between pasta and slirp4netns when it comes to creating/writing the hosts file (of course ips in the file might differ). We write the hosts file after the runtime creates the container (this is a recent change so this might be the cause for your chnage). Anyway running it as the `startContainer` hook should work. Of course at this moment you are already in the container namespace so there you need to open /etc/hosts directly.

**Reply by jontte75** [reply]
Thanks for the clarification! Indeed podman has also been updated lately to 4.6.2.
Now I just need to figure out what's the problem with our hook, because it just fails to execute the hook in stage startContainer:

`2023-09-26T10:25:09.000325689Z: error executing hook `/usr/libexec/k-oci-set-hosts` (exit code: 1)`

Perhaps some issue with namespace, since e.g. createRuntime and postStart seems to be executed nicely....need to re-check our hook implementation...
